tools/rpm2img

```diff
@@ -135,7 +135,34 @@ if [ "${PARTITION_PLAN}" == "split" ] ; then
     --sort --print "${DATA_IMAGE}"
 fi
 
+INSTALL_TIME="$(date -u +%Y-%m-%dT%H:%M:%SZ)"
 rpm -iv --root "${ROOT_MOUNT}" "${PACKAGE_DIR}"/*.rpm
+
+# inventory installed packages
+INVENTORY_QUERY="\{\"Name\":\"%{NAME}\"\
+,\"Publisher\":\"Bottlerocket\"\
+,\"Version\":\"${VERSION_ID}\"\
+,\"Release\":\"${BUILD_ID}\"\
+,\"InstalledTime\":\"${INSTALL_TIME}\"\
+,\"ApplicationType\":\"%{GROUP}\"\
+,\"Architecture\":\"%{ARCH}\"\
+,\"Url\":\"%{URL}\"\
+,\"Summary\":\"%{Summary}\"\}\n"
+
+mapfile -t installed_rpms <<< "$(rpm -qa --root "${ROOT_MOUNT}" \
+  --queryformat "${INVENTORY_QUERY}")"
+
+# shellcheck disable=SC2001
+if INVENTORY_DATA="$(jq --raw-output . \
+  <<< "$(sed "s/bottlerocket-${ARCH}-//g" <<< "${installed_rpms[@]}")" \
+  | jq --slurp 'sort_by(.Name)' | jq '{"Content": .}')"; then
+  printf "%s\n" "${INVENTORY_DATA}" > "${ROOT_MOUNT}/usr/share/bottlerocket/application-inventory.json"
+else
+  echo "something went wrong when generating the package inventory"
+  exit 1
+fi
```

Comment:
Can we rely on `set -e` instead?

```suggestion
# shellcheck disable=SC2001
INVENTORY_DATA="$(jq --raw-output . \
  <<< "$(sed "s/bottlerocket-${ARCH}-//g" <<< "${installed_rpms[@]}")" \
  | jq --slurp 'sort_by(.Name)' | jq '{"Content": .}')"
printf "%s\n" "${INVENTORY_DATA}" > "${ROOT_MOUNT}/usr/share/bottlerocket/application-inventory.json"
```

I'd also prefer putting the `sed` output into an intermediate variable just to simplify the invocation.